cmd/cmd.go

```diff
@@ -40,18 +40,24 @@ func NewPackCommand(logger ConfigurableLogger) (*cobra.Command, error) {
 		return nil, err
 	}
 
+	var forceColor bool
+
 	rootCmd := &cobra.Command{
 		Use:   "pack",
 		Short: "CLI for building apps using Cloud Native Buildpacks",
 		PersistentPreRun: func(cmd *cobra.Command, args []string) {
 			if fs := cmd.Flags(); fs != nil {
-				if flag, err := fs.GetBool("no-color"); err == nil && flag {
-					color.Disable(flag)
-				}
-
-				_, canDisplayColor := term.IsTerminal(logging.GetWriterForLevel(logger, logging.InfoLevel))
-				if !canDisplayColor {
-					color.Disable(true)
+				if forceColor {
+					color.Enabled()
```

Comment:
Hi @sarthaksarthak9 

If you use the `--force-color` flag, you will notice it will never reach this line. the variable declared on L43 is always set to `false` and you are never reading the flag value. 

Hint: check L53

Reply:
> Hi @sarthaksarthak9
> 
> If you use the `--force-color` flag, you will notice it will never reach this line. the variable declared on L43 is always set to `false` and you are never reading the flag value.
> 
> Hint: check L53

@jjbustamante Thank you, I will check this out!!

Reply:
I am planning to do this in this way !!! 
![Uploading Screenshot from 2023-11-10 08-21-11.png…]()

Reply:
Yes! just use the `fs.GetBool` function to validate if the flag was enabled by the end user. I think you don't need the `forceColor` flag anymore, right?

Reply:
Yah, that's true there is  no longer need to define --force-color explicitly as logic checks directly if the user has provided the --force-color flag using ` fs.GetBool `

Reply:
Wait! its fine to remove:

```go
var forceColor bool
```
but you still need:

```go
rootCmd.PersistentFlags().Bool("force-color", false, "Force color output")
```
remember, adding the flag, is the only way you have to allow users to set or not the flag, when the user set `--force-color` then `fs.GetBool("force-color")` will return true.

Also, you used to have:

```go
if force-color {
   color.Enabled() 
} else {
     if no-color { 
         color.Disable(true)
     } else {
         // more logic
     }
}
```

but you changed it to: 

```go
if force-color {
   color.Enabled() 
} 
if no-color { 
   color.Disable(true)
} else {
   // more logic
}
```

I think you will have troubles if run pack commands with both `--force-color` and `--no-color`, I know it is a crazy scenario, but I think the original flow is better.

Reply:
ohh, thanks a lot !! I will take care of this !!

Reply:
Hi @sarthaksarthak9. 

I apologized for all the back and forth, I think both of us are learning something about it!.

I was checking how the `color.Disable(bool)` and `color.Enable()` methods works, and I think we can do just one more adjust here.

By default, **color** is enabled, and we are just disabling it when `--no-color` or `is not a terminal`, I think what we can do is to do what we are doing today if `--force-color` is not the expected behavior. Something like:

```go
if forceColor, err := fs.GetBool("force-color"); err == nil && !forceColor {
	if flag, err := fs.GetBool("no-color"); err == nil && flag {
		color.Disable(flag)
	}

	_, canDisplayColor := term.IsTerminal(logging.GetWriterForLevel(logger, logging.InfoLevel))
	if !canDisplayColor {
		color.Disable(true)
	}
}


```

Reply:
okk, I will try my best